Quantum-Control-Applications/NV-Centers/Confocal-Setup/ramsey.py

```diff
@@ -0,0 +1,114 @@
+from qm.QuantumMachinesManager import QuantumMachinesManager
+from qm.qua import *
+import matplotlib.pyplot as plt
+from configuration import *
+
+###################
+# The QUA program #
+###################
+
+t_min = 4  # in clock cycles units (must be >= 4)
+t_max = 250  # in clock cycles units
+dt = 10  # in clock cycles units
+t_vec = np.arange(t_min, t_max + 0.1, dt)  # +0.1 to include t_max in array
+n_avg = 1e6
+
+with program() as ramsey:
+    counts1 = declare(int)  # saves number of photon counts
+    counts2 = declare(int)  # saves number of photon counts
+    times1 = declare(int, size=100)
+    times2 = declare(int, size=100)
+    counts_1_st = declare_stream()  # stream for counts
+    counts_2_st = declare_stream()  # stream for counts
+    t = declare(int)  # variable to sweep over in time
+    n = declare(int)  # variable to for_loop
+    n_st = declare_stream()  # stream to save iterations
+
+    play("laser_ON", "AOM")
+    wait(100)
+    with for_(n, 0, n < n_avg, n + 1):
+        with for_(t, t_min, t <= t_max, t + dt):
+            play("pi_half", "NV")  # Pi/2 pulse to qubit
+            wait(t, "NV")  # variable delay in spin Echo
+            play("pi_half", "NV")  # Pi/2 pulse to qubit
+
+            align()
+
+            play("laser_ON", "AOM")
+            measure("readout", "SPCM", None, time_tagging.analog(times1, meas_len, counts1))
+            save(counts1, counts_1_st)  # save counts
+            wait(100, "AOM")
+
+            align()
+
+            play("pi_half", "NV")  # Pi/2 pulse to qubit
+            wait(t, "NV")  # variable delay in spin Echo
+            frame_rotation_2pi(0.5, "NV")  # Turns next pulse to -x
+            play("pi_half", "NV")  # Pi/2 pulse to qubit
+            reset_frame("NV")
+
+            align()
+
+            play("laser_ON", "AOM")
+            measure("readout", "SPCM", None, time_tagging.analog(times2, meas_len, counts2))
+            save(counts2, counts_2_st)  # save counts
+            wait(100, "AOM")
+
+        save(n, n_st)  # save number of iteration inside for_loop
+
+    with stream_processing():
+        counts_1_st.buffer(len(t_vec)).average().save("counts1")
+        counts_2_st.buffer(len(t_vec)).average().save("counts2")
+        n_st.save("iteration")
+
+#####################################
+#  Open Communication with the QOP  #
+#####################################
+qmm = QuantumMachinesManager(host=qop_ip, port=qop_port)
+
+qm = qmm.open_qm(config)
+
+job = qm.execute(ramsey)  # execute QUA program
+
+res_handles = job.result_handles  # get access to handles
+counts1_handle = res_handles.get("counts1")
+counts2_handle = res_handles.get("counts2")
+iteration_handle = res_handles.get("iteration")
+counts1_handle.wait_for_values(1)
+counts2_handle.wait_for_values(1)
+iteration_handle.wait_for_values(1)
```

Comment:
I am waiting for Guy comments, but as soon as it is merged and ready we'll replace al of these by the new fetching tool

Reply:
Yes, definitely!